orga/app/components/campaign/charts/participants-by-mastery-percentage.js

```diff
@@ -0,0 +1,96 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import remove from 'lodash/remove';
+import sumBy from 'lodash/sumBy';
+
+export default class ParticipantsByStage extends Component {
+  @service store;
+  @service intl;
+
+  @tracked data = [];
+  @tracked loading = true;
+
+  constructor(...args) {
+    super(...args);
+    const { campaignId } = this.args;
+
+    const adapter = this.store.adapterFor('campaign-stats');
+    adapter
+      .getParticipationsByMasteryRate(campaignId)
+      .then((response) => {
+
+        const { steps, labels } = _buildChartDatas(response.data.attributes['result-distribution']);
+
+        this.data = {
+          labels,
+          datasets: [
+            {
+              data: steps,
+              border: '#3D68FF',
+              backgroundColor: '#3D68FF',
+            },
+          ],
+        };
+
+        this.loading = false;
+      });
+  }
+
+  get options() {
+    return {
+      animation: false,
+      responsive: true,
+      maintainAspectRatio: false,
+      scales: {
+        yAxis: {
+          type: 'linear',
+          beginAtZero: true,
+          grid: {
+            borderDash: [4, 4],
+          },
+          ticks: {
+            stepSize: 1,
+          },
+        },
+        xAxis: {
+          grid: {
+            display: false,
+          },
+        },
+      },
+      plugins: {
+        legend: {
+          display: false,
+        },
+        tooltip: {
+          callbacks: {
+            title: (tooltipItem) => {
+              return this.intl.t('charts.participants-by-mastery-percentage.tooltip.title', { legend: tooltipItem[0].label });
+            },
+            label: (data) => {
+              return this.intl.t('charts.participants-by-mastery-percentage.tooltip.label', { count: data.raw });
+            },
+          },
+        },
+      },
+    };
+  }
+}
+
+function _buildChartDatas(resultDistributions) {
+  const steps = [];
+  const labels = [];
+
+  for (let i = 10; i <= 100; i = i + 10) {
+    if (i === 10) {
+      labels.push('0% - 10%');
+    } else {
+      labels.push(`${i - 9}% - ${i}%`);
```

Comment:
labels.push(t('charts.participants-by-mastery-percentage.label', { from: i - 9, to: i });

Reply:
done